src/Features/CSharp/Portable/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -120,8 +120,10 @@ protected override IPattern<CasePatternSwitchLabelSyntax> CreatePatternFromExpre
                             // separate out the leftmost expression and reconstruct the rest of conditions into a new
                             // expression, so that they would be evaluated entirely after "expr" e.g. "expr && (cond1 && cond2)".
                             // Afterwards, we can directly use it in a case guard e.g. "case pat when cond1 && cond2:".
-                            var condition = (leftmost.Parent.Parent as BinaryExpressionSyntax)
-                                ?.WithLeft(((BinaryExpressionSyntax)leftmost.Parent).Right);
+                            var parentBinary = leftmost.GetAncestor<BinaryExpressionSyntax>();
```

Comment:
❔ The use of `GetAncestor` makes me uncomfortable (things like nested blocks/lambdas are difficult to reason about). Why not `WalkUpParentheses`?